lightning/src/ln/chan_utils.rs

```diff
@@ -25,6 +25,25 @@ use secp256k1;
 pub(super) const HTLC_SUCCESS_TX_WEIGHT: u64 = 703;
 pub(super) const HTLC_TIMEOUT_TX_WEIGHT: u64 = 663;
 
+#[derive(PartialEq)]
+pub(super) enum HTLCType {
+	AcceptedHTLC,
+	OfferedHTLC,
+	NoneHTLC,
+}
+
+impl HTLCType {
+	pub(super) fn weight_to_htlctype(witness_script_len: usize) ->  HTLCType {
```

Comment:
This should return an Option instead of having a NoneHTLC enum

Reply:
I disagree a bit here, I don't found it makes callsites clearer, like "else if (if let Some(htlc_type) = weight_to_htlctype(script.len()) { htlc_type == HTLCType::Offered } else { false })"

Reply:
You should be able to make the callsets else if weight_to_htlctype(script.len())  == Some(HTLCType::Offered), which reads fine to me?

Reply:
My periodic reminder of Rust greatness.